### PR TITLE
SOLR-16174: Modernize TestBulkSchemaConcurrent

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16174

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java [hunk 1]
```diff
@@ -51,39 +60,50 @@ protected String getCloudSolrConfig() {
...
-
-      threads[i].start();
+      callees.add(call);
     }
 
-    for (Thread thread : threads) thread.join();
+    executorService.invokeAll(callees);
+    executorService.shutdown();
+    assertTrue(
+        "Running for too long...", executorService.awaitTermination(TIMEOUT * 3, TimeUnit.SECONDS));
```
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java:106]: Yea I'll add a comment here. I'm pretty sure the logic is correct to be 3*TIMEOUT due to 

```
invokeBulkAddCall(finalI, errs);
invokeBulkReplaceCall(finalI, errs);
invokeBulkDeleteCall(finalI, errs);
```

with each method looping the length of the timeout?

File: solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java [hunk 2]
```diff
@@ -51,39 +60,50 @@ protected String getCloudSolrConfig() {
...
+          () -> {
+            List<String> errs = new ArrayList<>();
+            collectErrors.add(errs);
+            try {
+              invokeBulkAddCall(finalI, errs);
+              invokeBulkReplaceCall(finalI, errs);
+              invokeBulkDeleteCall(finalI, errs);
+            } catch (Exception e) {
+              e.printStackTrace();
+              Thread.currentThread().interrupt();
```
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java:96]: Yea this is leftover from trying to figure out why things were hanging. I'll fix it to just interruptedexception.

File: solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java [hunk 3]
```diff
@@ -51,39 +60,50 @@ protected String getCloudSolrConfig() {
...
+      Callable<Void> call =
+          () -> {
+            List<String> errs = new ArrayList<>();
+            collectErrors.add(errs);
+            try {
+              invokeBulkAddCall(finalI, errs);
+              invokeBulkReplaceCall(finalI, errs);
+              invokeBulkDeleteCall(finalI, errs);
+            } catch (Exception e) {
+              e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java:95]: I put a comment in here and changed it to use a logger. I'm not sure this error would be logged later in every case.

File: solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java [hunk 4]
```diff
@@ -340,9 +370,13 @@ private void invokeBulkDeleteCall(int seed, ArrayList<String> errs) throws Excep
 
   private boolean checkCopyField(
       @SuppressWarnings({"rawtypes"}) List<Map> l, String src, String dest) {
-    if (l == null) return false;
+    if (l == null) {
+      return false;
+    }
     for (@SuppressWarnings({"rawtypes"}) Map map : l) {
```
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java]: addressed in 9389c95304d27d3b4308bc974e47df2260582ccf

File: solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java [hunk 5]
```diff
@@ -51,39 +60,50 @@ protected String getCloudSolrConfig() {
...
+      callees.add(call);
     }
 
-    for (Thread thread : threads) thread.join();
+    executorService.invokeAll(callees);
+    executorService.shutdown();
+    assertTrue(
+        "Running for too long...", executorService.awaitTermination(TIMEOUT * 3, TimeUnit.SECONDS));
 
     boolean success = true;
```
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java:111]: addressed in 9389c95304d27d3b4308bc974e47df2260582ccf

File: solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java:93]: addressed in 9389c95304d27d3b4308bc974e47df2260582ccf

File: solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java]: addressed in 9389c95304d27d3b4308bc974e47df2260582ccf

File: solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java:108]: addressed in 9389c95304d27d3b4308bc974e47df2260582ccf

File: solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java [hunk 6]
```diff
@@ -51,39 +60,50 @@ protected String getCloudSolrConfig() {
     return "solrconfig-managed-schema.xml";
   }
 
+  @Before
+  public void setup() {
+    super.beforeTest();
```
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java]: addressed in 9389c95304d27d3b4308bc974e47df2260582ccf

File: solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java [hunk 7]
```diff
@@ -145,46 +166,49 @@ private void invokeBulkAddCall(int seed, ArrayList<String> errs) throws Exceptio
...
-                  "CopyField source={0},dest={1} not created", aField, dynamicCopyFldDest));
-
-        m = getObj(harness, newFieldTypeName, "fieldTypes");
-        if (m == null)
-          errmessages.add(StrUtils.formatString("new type {0}  not created", newFieldTypeName));
-
-        if (errmessages.isEmpty()) break;
-
-        Thread.sleep(10);
+    long startTime = System.nanoTime();
```
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java:172]: @madrob do you have an example? I'm not sure what you mean by `TimeOut` here?

[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java:172]: Thanks - didn't realize there was a Solr `TimeOut` function. I was looking for a JDK one :D

[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/test/org/apache/solr/schema/TestBulkSchemaConcurrent.java]: Addressed in b5d268ef8eb045c703fbcdef1e8987ebbbf061da

**Review by madrob — APPROVED** [review]

**Pull request merged**